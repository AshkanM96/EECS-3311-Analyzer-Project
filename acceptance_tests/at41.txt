open_set_enumeration
boolean_constant(true)
boolean_constant(true)
boolean_constant(true)
boolean_constant(true)
type_check
close_set_enumeration
evaluate
type_check
reset
divide
integer_constant(0)
integer_constant(5)
type_check
evaluate
reset
divide
integer_constant(10)
sum
open_set_enumeration
integer_constant(5)
integer_constant(5)
negative
integer_constant(3)
close_set_enumeration
type_check
evaluate
