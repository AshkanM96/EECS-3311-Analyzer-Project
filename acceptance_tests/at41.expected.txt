  Expression currently specified: ?
  Report: Expression is initialized.
->open_set_enumeration
  Expression currently specified: {?}
  Report: OK.
->boolean_constant(True)
  Expression currently specified: {True, ?}
  Report: OK.
->boolean_constant(True)
  Expression currently specified: {True, True, ?}
  Report: OK.
->boolean_constant(True)
  Expression currently specified: {True, True, True, ?}
  Report: OK.
->boolean_constant(True)
  Expression currently specified: {True, True, True, True, ?}
  Report: OK.
->type_check
  Expression currently specified: {True, True, True, True, ?}
  Report: Error (Expression is not yet fully specified).
->close_set_enumeration
  Expression currently specified: {True, True, True, True}
  Report: OK.
->evaluate
  Expression currently specified: {True, True, True, True}
  Report: {True}
->type_check
  Expression currently specified: {True, True, True, True}
  Report: {True, True, True, True} is type-correct.
->reset
  Expression currently specified: ?
  Report: OK.
->divide
  Expression currently specified: (? / nil)
  Report: OK.
->integer_constant(0)
  Expression currently specified: (0 / ?)
  Report: OK.
->integer_constant(5)
  Expression currently specified: (0 / 5)
  Report: OK.
->type_check
  Expression currently specified: (0 / 5)
  Report: (0 / 5) is type-correct.
->evaluate
  Expression currently specified: (0 / 5)
  Report: 0
->reset
  Expression currently specified: ?
  Report: OK.
->divide
  Expression currently specified: (? / nil)
  Report: OK.
->integer_constant(10)
  Expression currently specified: (10 / ?)
  Report: OK.
->sum
  Expression currently specified: (10 / (+ ?))
  Report: OK.
->open_set_enumeration
  Expression currently specified: (10 / (+ {?}))
  Report: OK.
->integer_constant(5)
  Expression currently specified: (10 / (+ {5, ?}))
  Report: OK.
->integer_constant(5)
  Expression currently specified: (10 / (+ {5, 5, ?}))
  Report: OK.
->negative
  Expression currently specified: (10 / (+ {5, 5, (- ?)}))
  Report: OK.
->integer_constant(3)
  Expression currently specified: (10 / (+ {5, 5, (- 3), ?}))
  Report: OK.
->close_set_enumeration
  Expression currently specified: (10 / (+ {5, 5, (- 3)}))
  Report: OK.
->type_check
  Expression currently specified: (10 / (+ {5, 5, (- 3)}))
  Report: (10 / (+ {5, 5, (- 3)})) is type-correct.
->evaluate
  Expression currently specified: (10 / (+ {5, 5, (- 3)}))
  Report: 5
